File: Graph Algorithms in Genome Sequencing/Genome seq/Construct the de Bruijn Graph of a String/DeBruijnString.java
import java.util.Scanner;

public class DeBruijnString {
    /*
     * Creates a de Bruijn graph from a string given a value of k
     *
     * @param k         the length of strings represented by edges in the 
     *                  de Bruijn graph
     * @param text      the String from which the de Bruijn graph will be 
     *                  constructed
     *
     * @return a String containing an adjacency list representation of the
     * de Bruijn graph as described in the problem specification
     */
    public static String deBruijn(int k, String text) {
        // TODO: your code here
        return "";
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);

        int k = stdin.nextInt();
        stdin.nextLine(); // skip newline after k

        String text = stdin.nextLine();

        System.out.println(deBruijn(k, text));
    }
}

File: Graph Algorithms in Genome Sequencing/Genome seq/Construct the de Bruijn Graph of a String/de_bruijn_string.cpp
#include <iostream>
#include <vector>
#include <string>

using std::cout;
using std::cin;
using std::endl;
using std::string;
using std::vector;


/*
 * Creates a de Bruijn graph from a string given a value of k
 *
 * \param k         the length of strings represented by edges in the 
 *                  de Bruijn graph
 * \param text      the string from which the de Bruijn graph will be 
 *                  constructed
 *
 * \return a string containing an adjacency list representation of the
 * de Bruijn graph as described in the problem specification
 */
string de_bruijn(int k, const string& text) {
    // TODO: your code here
    return "";
}


int main(int argc, char* argv[]) {
    int k;
    cin >> k;

    string text;
    cin >> text;

    cout << de_bruijn(k, text) << endl;
}
